### PR TITLE
[bitnami/kafka] change zookeeper keystore pem location

### PR DESCRIPTION
**Pull request by toondaey**
### Description of the change

I think this necessary because from experience, I find that I keep having to update my local keystore which somehow became more of a burden

### Benefits

Eliminate the need to update and change zookeeper.keystore.pem every single time.

### Possible drawbacks

None that I can think of.

### Applicable issues

Frustrating need to keep updating local zookeeper.keystore.pem evicted.

### CODE REVIEW

File: bitnami/kafka/3.2/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh [hunk 1]
```diff
@@ -772,8 +772,8 @@ kafka_zookeeper_configure_tls() {
     elif [[ "$KAFKA_ZOOKEEPER_TLS_TYPE" = "PEM" ]] && [[ -f "$KAFKA_CERTS_DIR"/zookeeper.keystore.pem ]] && [[ -f "$KAFKA_CERTS_DIR"/zookeeper.keystore.key ]]; then
         # Concatenating private key into public certificate file
         # This is needed to load keystore from location using PEM
-        cat "$KAFKA_CERTS_DIR"/zookeeper.keystore.key >>"$KAFKA_CERTS_DIR"/zookeeper.keystore.pem
-        keystore_location="${KAFKA_CERTS_DIR}/zookeeper.keystore.pem"
+        cat "$KAFKA_CERTS_DIR"/zookeeper.keystore.pem "$KAFKA_CERTS_DIR"/zookeeper.keystore.key > "$KAFKA_CERTS_DIR"/zookeeper-keypair.pem
+        keystore_location="$KAFKA_CERTS_DIR"/zookeeper-keypair.pem
```
Comment (andresbono) [inline bitnami/kafka/3.2/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh:776]: What do you think about this minor suggestion?

```suggestion
        keystore_location="${KAFKA_CERTS_DIR}/zookeeper.keypair.pem"
        cat "${KAFKA_CERTS_DIR}/zookeeper.keystore.pem" "${KAFKA_CERTS_DIR}/zookeeper.keystore.key" > "$keystore_location"
```

Could you apply it to the rest of the files?

**Review by andresbono — COMMENTED** [review]
LGTM, thank you!!  The current implementation would also be a problem if the file is mounted as a read-only secret.

 I just added a minor suggestion about the style. Could you please check it?

**Comment by toondaey** [comment]
> LGTM, thank you!! The current implementation would also be a problem if the file is mounted as a read-only secret.
> 
> I just added a minor suggestion about the style. Could you please check it?

Actually that might not because I've you can use `cat` with a readonly mount into container files. In my local docker setup, I mounted all the other certificates as `ro` save for those of zookeeper because of the modifications.

I've applied and made some changes to the modifications.

File: bitnami/kafka/3.2/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh [hunk 2]
```diff
@@ -772,8 +772,8 @@ kafka_zookeeper_configure_tls() {
     elif [[ "$KAFKA_ZOOKEEPER_TLS_TYPE" = "PEM" ]] && [[ -f "$KAFKA_CERTS_DIR"/zookeeper.keystore.pem ]] && [[ -f "$KAFKA_CERTS_DIR"/zookeeper.keystore.key ]]; then
         # Concatenating private key into public certificate file
         # This is needed to load keystore from location using PEM
-        cat "$KAFKA_CERTS_DIR"/zookeeper.keystore.key >>"$KAFKA_CERTS_DIR"/zookeeper.keystore.pem
-        keystore_location="${KAFKA_CERTS_DIR}/zookeeper.keystore.pem"
+        keystore_location="${KAFKA_CERTS_DIR}/zookeeper.keypair.pem"
+        cat "${KAFKA_CERTS_DIR}/zookeeper.keystore.{pem,key}" > "$keystore_location"
```
Comment (andresbono) [inline bitnami/kafka/3.2/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh:776]: Unfortunately, [brace expansion](https://www.gnu.org/software/bash/manual/html_node/Brace-Expansion.html) will not work in a quoted string.

In the repository, we tend to favor the style of variable interpolation with curly braces and quoting the whole string, for example `"${var}/foo/bar"`. Although you can find exceptions. In this case, following this style convention would prevent brace expansion from working. So we need:

1)
```shell
cat "$KAFKA_CERTS_DIR"/zookeeper.keystore.{pem,key} > "$keystore_location"
```

or

2)
```shell
cat "${KAFKA_CERTS_DIR}/zookeeper.keystore.pem" "${KAFKA_CERTS_DIR}/zookeeper.keystore.key" > "$keystore_location"
```

I would go with the second option to follow the common style convention, but option one will also work.

**Review by andresbono — COMMENTED** [review]
> > LGTM, thank you!! The current implementation would also be a problem if the file is mounted as a read-only secret.
> > I just added a minor suggestion about the style. Could you please check it?
> 
> Actually that might not because I've you can use `cat` with a readonly mount into container files. In my local docker setup, I mounted all the other certificates as `ro` save for those of zookeeper because of the modifications.

By "current implementation" I meant the implementation before your PR. I was just highlighting that `zookeeper.keystore.pem` could not be mounted as `ro` because it needed to be writable. With your PR you are also fixing that. Is that correct?

Please, check my last in-line comment. Thank you!

File: bitnami/kafka/3.2/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh (hunk 2 above)
Comment (toondaey) [reply bitnami/kafka/3.2/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh:776]: Yes, this should fix the current implementation. Apologies for not checking for the common style. I'll stick with the second one as well.

**Comment by toondaey** [comment]
I hope this is ok? I discovered that `KAFKA_KRAFT_CLUSTER_ID` had a typo [here](https://github.com/bitnami/containers/blob/896caff5aedf91ce379effe71ac3b2ca5e9f655f/bitnami/kafka/README.md?plain=1#L210)

File: bitnami/kafka/3.2/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh (hunk 2 above)
Comment (andresbono) [reply bitnami/kafka/3.2/debian-11/rootfs/opt/bitnami/scripts/libkafka.sh]: No problem at all!! Thank you for adding the suggestion.

**Review by andresbono — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**